application/helpers/previewer_helper.php

```diff
@@ -139,39 +130,100 @@ function create_title_slug($project)
 
 }
 
-
-// builds link to catalog author's page
-function build_author_link($author, $wiki='', $link='')
+function get_language_code($language)
 {
+	if (empty($language)) return '';
 
-	if (empty($author)) return '';
+	return (empty($language->two_letter_code)) ? $language->three_letter_code: $language->two_letter_code;
 
-	$link = (empty($link))? base_url().'author/'. $author->id : $link;
+}
 
-	return '<a href="'.$link.'">'. $wiki . build_author_name($author) .'</a>';
+function format_author_name($author)
+{
+	if (isset($author->author))
+		return trim($author->author);
+	else
+		return implode(' ', array_filter(array($author->first_name, $author->last_name)));
 }
 
-function build_author_name($author)
+function format_author_years($author)
 {
-	return implode(' ', array_filter(array($author->first_name, $author->last_name)));
+	$dob = empty($author->dob) ? '' : $author->dob;
+	$dod = empty($author->dod) ? '' : $author->dod;
+
+	if (!empty($dod) || !empty($dod))
+		return '(' . $dob . ' - ' . $dod . ')';
+	else
+		return '';
 }
 
-function get_language_code($language)
+define ('FMT_AUTH_YEARS', 0x01);
+define ('FMT_AUTH_HTML',  0x02);
+define ('FMT_AUTH_LINK',  0x04);
+define ('FMT_AUTH_WIKI',  0x08);
+
+function format_authors($authors, $flags = 0, $max = 0)
 {
-	if (empty($language)) return '';
+	$flag_years = (bool)($flags & FMT_AUTH_YEARS);
```

Comment:
This is such a C thing to do ;)

Reply:
>This is such a C thing to do ;)

Is that a bad thing? Or a good thing?

I actually got the idea partially from PHP functions like https://www.php.net/manual/en/function.mb-convert-case.php which use defined constants to pass in different options. But I just looked again and I see you can only pass in _one_ constant to that function. 

I wonder if any standard PHP functions use an ORed options mask like I used? 

Is there a commonly accepted way to do something similar in PHP?

Languages like Perl often use a hash to pass in various options like this. I guess you could use an (associative) array to do something similar in PHP. But when you learned to program on a computer with 48kB of RAM these methods just don't sit well with me...  they cause global warming.

Reply:
> > This is such a C thing to do ;)
> 
> Is that a bad thing? Or a good thing?

It's just... a thing :) It shows your origins, and experience, but is neither good nor bad. 

> I actually got the idea partially from PHP functions like https://www.php.net/manual/en/function.mb-convert-case.php which use defined constants to pass in different options. But I just looked again and I see you can only pass in _one_ constant to that function.
> 
> I wonder if any standard PHP functions use an ORed options mask like I used?
> 
> Is there a commonly accepted way to do something similar in PHP?

I don't know PHP well enough to have an informed opinion. I think in Python we'd just have a bunch of boolean keyword arguments with default values, something like:

```python
def format_authors(authors, years=False, html=False, link=False, wiki=False):
```

I believe something similar is possible with PHP.

> Languages like Perl often use a hash to pass in various options like this. I guess you could use an (associative) array to do something similar in PHP. But when you learned to program on a computer with 48kB of RAM these methods just don't sit well with me... they cause global warming.

I... don't have an answer to that. Or maybe I do, but it's very long and philosophical, and involves how my kids grow up in a world where "computers" are for solely consuming media rather than tinkering and creating or programming, and is more appropriate for a pub than a pull request.

Reply:
>def format_authors(authors, years=False, html=False, link=False, wiki=False):

That is possible in PHP as well. But the annoying thing with that is you need to remember the order of the parameters when using the function add add a bunch of extra false parameter values to use the last option.

Anyway... that is probably enough somewhat off topic discussion in this PR.